### ISSUE TITLE
Potential Support for UWP?

### ISSUE THREAD

**Issue by theweavrs** [rpgmaker/NetJSON#167]
Is it possible to support UWP? I would love having such a fast JSON serializer/deserializer on UWP.

**Comment by rpgmaker** [comment]
Does UWP support .net core on standard 1.6? If it does then it should already work.  Beside that, the only way is to support portable class library. it is difficult because it missing so much of the reflection.emit class that to make it work I will have to make a non reflection emit version of it using something like expression tree(Maybe expression tree given in direct access to reflection emit) or delegate which will be far slower than the current version.

So far I have received enough support for usage of it in UWP. But I guess I can always revisit it and see what is possible and get back to you.

Thanks,

**Comment by Nexxkinn** [comment]
In the official documentation UWP only supports .Net Standard until version 1.4, which means that you can't import the library into UWP.
https://docs.microsoft.com/en-us/dotnet/articles/standard/library

On the .NET Core Roadmap, UWP will support .Net Standard 2.0 when the Fall Creators Update released.
https://github.com/dotnet/core/blob/master/roadmap.md

In the other hand, You can start compile it to .Net Standard 2.0 now, since the preview version got released days ago.

In conclusion : Expect Microsoft to release a preview SDK few months before the update publicly released.

**Comment by rpgmaker** [comment]
Can I close this assuming netstandard 2.0 will automatically means that UWP get support in the future? Will close it next week if I don't get any reply.

Thanks,

**Comment by imacks** [comment]
I was looking at the same problem, but I'm trying to target NETSTANDARD1.3.

It seems that DependencyContext and GetEntryAssembly are the only reason we need to be on NETSTANDARD1.5. So I did something really lame:

```C#
#if NETSTANDARD && !(NETSTANDARD1_5 || NETSTANDARD1_6)
		private static Assembly _entryAssembly;
		private static Assembly[] _loadedAssembly;
#endif

		public static Assembly EntryAssembly
		{
			get 
			{ 
#if NETSTANDARD1_5 || NETSTANDARD1_6
				return Assembly.GetEntryAssembly(); 
#elif NETSTANDARD && !(NETSTANDARD1_5 || NETSTANDARD1_6)
				return _entryAssembly; 
#else
				return Assembly.GetExecutingAssembly();
#endif
			}
			set 
			{ 
#if NETSTANDARD && !(NETSTANDARD1_5 || NETSTANDARD1_6)
				_entryAssembly = value; 
#else
				return;
#endif
			}
		}

		public static Assembly[] LoadedAssemblies
		{
			get 
			{ 
#if NETSTANDARD1_5 || NETSTANDARD1_6
				return DependencyContext.Default.GetDefaultAssemblyNames().Select(x => Assembly.Load(x)).ToArray();
#elif NETSTANDARD && !(NETSTANDARD1_5 || NETSTANDARD1_6)
				return _loadedAssembly; 
#else
				return AppDomain.CurrentDomain.GetAssemblies();
#endif
			}
			set 
			{ 
#if NETSTANDARD && !(NETSTANDARD1_5 || NETSTANDARD1_6)
				_loadedAssembly = value; 
#else
				return;
#endif
			}
		}

        internal static MethodInfo DefineMethodEx(this TypeBuilder builder, string methodName, MethodAttributes methodAttribute, Type returnType, Type[] parameterTypes)
        {
			if (builder == null)
				return new DynamicMethod(methodName, returnType, parameterTypes, EntryAssembly.ManifestModule, true);

            return builder.DefineMethod(methodName, methodAttribute, returnType, parameterTypes);
        }

		private static List<Type> GetIncludedTypeTypes(Type type)
		{
			[...]
			foreach (var asm in LoadedAssemblies)
			[...]
		}
```

A bit of polyfills (mainly for `GetInterfaces` if I recall) and it will happily compile on NETStandard1.3, though I need to manually set the `EntryAssembly` and `LoadedAssemblies` properties before using `NETJson.Serialize`.

**Comment by rpgmaker** [comment]
Interesting solution. I tried a polyfill solution before, but never like how you did it. Maybe you can hack and use the filename and method name from the attribute that allows you to get compile time info of your code such as callermemberAttribute or something close to auto discover the assembly you need.

**Comment by rpgmaker** [comment]
Since the solution to this problem is mostly custom for a specific version of net standard. I suggest you create it as a pr contribution once you get it working.

Feel free to reopen this if you want to link the PR to this.

Thanks,
